### ISSUE TITLE
Validate common context properties

### ISSUE THREAD

**Issue by flcdrg** [rhysd/actionlint#39]
I just found a bug in a workflow because I was trying to reference this property:

```yaml
${{ github.event.worflow_run.head_commit.message }}
```

It was only after I copy and pasted the names from the output of `${{ toJSON(github) }}` that I realised I'd made an embarrasing typo -  (it should be `workflow_run`)!

What I'm wondering is whether actionlint could catch these kinds of things? How deep you could validate is a good question, but I'd say at least `github.event.xxxxx` would be a starting point, where xxxxx is one of the events listed at https://docs.github.com/en/actions/reference/events-that-trigger-workflows

**Comment by rhysd** [comment]
Good point. actionlint currently types `github.event` as an object which accepts *any* properties.

It's possible to give stricter type to `github.event`. But it is a bit challenging in the following points:

- there are so many properties under `github.event`. Crafting the type definition is a bit hard.
- The event payloads are depending on GitHub API specification, meant that they would change at some point in the future.

My idea is preparing a script to generate a Go code for the type of `github.event` and generating a code regularly. We already do this for [popular_actions.go](https://github.com/rhysd/actionlint/tree/main/scripts/generate-popular-actions) and [all_webhooks.go](https://github.com/rhysd/actionlint/tree/main/scripts/generate-webhook-events). I think we can do the similar thing for the type definition of `github.event` payload.

**Comment by rhysd** [comment]
Properties other than `github.event.*` are already typed strictly following the official document. So I believe the remaining payloads which are not typed well are only `github.event.*`. Let me update the title more specifically.
